dotCMS/src/main/java/com/dotcms/rest/api/v1/diy/DIYResource.java

```diff
@@ -0,0 +1,98 @@
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.IdentifierAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.fileassets.business.FileAsset;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+@Path("/v1/diy")
+public class DIYResource {
+
+    private final WebResource webResource = new WebResource();
+    private final String DIYPath = "/diy";
+    private final HostAPI hostAPI;
+    private final IdentifierAPI identifierAPI;
+    private final ContentletAPI contentletAPI;
+    private final String GET_FILE_NAME = "get";
+    private final String FILE_EXTENSION = ".vtl";
+
+    public DIYResource() {
+        this(APILocator.getHostAPI(), APILocator.getIdentifierAPI(), APILocator.getContentletAPI());
+    }
+
+    @VisibleForTesting
+    DIYResource(final HostAPI hostAPI, final IdentifierAPI identifierAPI, final ContentletAPI contentletAPI) {
+        this.hostAPI = hostAPI;
+        this.identifierAPI = identifierAPI;
+        this.contentletAPI = contentletAPI;
+    }
+
+    @GET
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    @Path("/{folder}/{path: .*}")
+    public Response get(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                        @Context UriInfo uriInfo, @PathParam("folder") final String folderName,
+                             @PathParam("path") final String pathParams) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (pathParams, true, request, true, null);
+
+        final User user = initDataObject.getUser();
+        final Language currentLanguage = WebAPILocator.getLanguageWebAPI().getLanguage(request);
+
+        MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
+
+        Map<String, String> dotJSON;
+
+        try {
+            Host site = this.hostAPI.resolveHostName(request.getServerName(), user, false);
+            final String getFilePath = DIYPath + "/" + folderName + "/" + GET_FILE_NAME + FILE_EXTENSION;
```

Comment:
this should be a StringBuilder
```
new StringBuilder().append(DIYPath).append(StringPool.SLASH).append(folderName).append(StringPool.SLASH).append(GET_FILE_NAME).append(FILE_EXTENSION).toString();

Reply:
In general I would prefer to have a resolver that are not tied to the vtl in order to support future languages such as Javascript, something like:

```
final  FileAsset getFileAsset = this.resourceResolver.resolve (DIY_PATH, folderName, RequestMethod.GET);
final Map<String, Object> context = map("initDataObject", initDataObject,  // if you want the user.
"queryParams", queryParameters,"dotJSON", new HashMap())
final ScriptingStrategy strategy = this.scriptingStrategyFactory.create (getFileAsset); // this should be cache
try {
    strategy.pre (context);  // throws errors in case invalid
    strategy.handle (context);
    //....
} finally {
    strategy.post(context);
}
```